submittal/src/main/java/com/kyriba/submittal/service/SpeakerServiceImpl.java

```diff
@@ -0,0 +1,147 @@
+package com.kyriba.submittal.service;
+
+import com.kyriba.submittal.dao.RegistrationRepository;
+import com.kyriba.submittal.dao.SpeakerRepository;
+import com.kyriba.submittal.dao.TopicRepository;
+import com.kyriba.submittal.domain.Registration;
+import com.kyriba.submittal.domain.Speaker;
+import com.kyriba.submittal.domain.SpeakerStatus;
+import com.kyriba.submittal.domain.Topic;
+import com.kyriba.submittal.domain.dto.JobInfo;
+import com.kyriba.submittal.domain.dto.PersonalInfo;
+import com.kyriba.submittal.domain.dto.TopicDto;
+import com.kyriba.submittal.domain.dto.TopicInfo;
+import com.kyriba.submittal.service.exception.DuplicateObjectException;
+import com.kyriba.submittal.service.exception.ObjectNotFoundException;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.util.Pair;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.List;
+import java.util.stream.StreamSupport;
+
+import static java.util.stream.Collectors.toList;
+
+
+/**
+ * @author M-ABL
+ */
+@Service
+@Transactional
+@RequiredArgsConstructor
+public class SpeakerServiceImpl implements SpeakerService
+{
+  private final SpeakerRepository speakerRepository;
+  private final TopicRepository topicRepository;
+  private final RegistrationRepository registrationRepository;
+  private final KeyGenerator keyGenerator;
+
+
+  @Override
+  public Pair<Long, List<Long>> register(final String email,
+                                         final String password,
+                                         final PersonalInfo personalInfo,
+                                         final JobInfo jobInfo,
+                                         final List<TopicInfo> topics,
+                                         final long conferenceId)
+  {
+    if (speakerRepository.findByEmail(email).isPresent()) {
+      throw new DuplicateObjectException("Email is already in use");
+    }
+    Speaker speaker = new Speaker();

```

Comment:
When replace input params with DTO create constructor or factory method on entity to create from DTO.

Reply:
then I have to move com.kyriba.submittal.api.SpeakerController.SpeakerRequest and com.kyriba.submittal.api.SpeakerController.TopicRequest to the dto package...

Reply:
It is completely OK. It is much better to transfer full set of data in DTO instead of passing N parameters.